plugins/network-elements/nuage-vsp/src/com/cloud/network/manager/NuageVspManagerImpl.java

```diff
@@ -348,6 +866,6 @@ public String getConfigComponentName() {
 
     @Override
     public ConfigKey<?>[] getConfigKeys() {
-        return new ConfigKey<?>[] {NuageVspSyncWorkers, NuageVspSyncInterval};
+        return NUAGE_VSP_CONFIG_KEYS;
```

Comment:
Perform an ``Arrays.copyOf()`` before returning ``NUAGE_VSP_CONFIG_KEYS`` to preventing side effects in the event a subsequent user attempts to modify the array.